das/daser.go

```
@@ -71,7 +56,15 @@ func NewDASer(
 		subscriber:     newSubscriber(),
 		subscriberDone: make(chan struct{}),
 	}
-	d.sampler = newSamplingCoordinator(concurrencyLimit, samplingRange, getter, d.sample)
+
+	for _, applyOpt := range options {
+		err := applyOpt(d)
+		if err != nil {
+			panic(err)
```

Let's return an error here.